src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -209,6 +213,40 @@ public IndexFeatureSet indexFeatureSet()
         return indexFeatureSet;
     }
 
+    /**
+     * Returns whether this query is selection the {@link PrimaryKey}.
+     * The query selects the key if:
+     *  1. The query is not row-aware
+     *  2. The table associated with the query is not using clustering keys
+     *  3. The clustering index filter for the command wants the row.
+     *
+     *  Item 3 is important in paged queries where the {@link org.apache.cassandra.db.filter.ClusteringIndexSliceFilter} for
+     *  subsequent paged queries may not select rows that are returned by the index
+     *  search because that is initially partition based.
+     *
+     * @param key The {@link PrimaryKey} to be tested
+     * @return true if the key is selected by the query
```

Comment:
This is a better description of the method. Just make it a top-level description and remove `@return`

Reply:
I'm not sure I follow this last bit but I've updated the top-level description.